### PR TITLE
Hrm05 heart rate measure

### PR DESCRIPTION
**Pull request by akshaya86**

### CODE REVIEW

File: app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    private fun stopMeasureHRView(){
+        stopAnimation()
+        binding.clHeartStartId.visibility = View.VISIBLE
+        binding.clHeartStopId.visibility = View.GONE
+        mSensorManager?.unregisterListener(this)
+    }
+
+    private fun getSensorData() {
+        mSensorManager = getSystemService(SENSOR_SERVICE) as SensorManager
+        mHeartSensor = mSensorManager?.getDefaultSensor(Sensor.TYPE_HEART_RATE)
```
Comment (EntyR) [inline app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt]: Are you able to get mocked data from emulator this way? Genially you should use [health service api](https://developer.android.com/training/wearables/health-services) when trying to receive heart rate events. but emulator have some problem providing this values. That's why you allowed to use any other sensor

Comment (EntyR) [inline app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt]: Are you able to get mocked data from emulator this way? Genially you should use [health service api](https://developer.android.com/training/wearables/health-services) when trying to receive heart rate events. but emulator have some problem providing this value

File: app/src/main/java/com/example/wearableapp/presentation/service/HeartRateSensorService.kt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            val manager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
+            manager.createNotificationChannel(channel)
+        }
+
+        val notification = NotificationCompat.Builder(this, SENSOR_ID)
+            .setContentTitle(this.getString(R.string.sensor_title))
+            .setSmallIcon(R.drawable.ic_heart_full_icon)
+            .setContentText(getString(R.string.sensor_desc))
+            .setContentIntent(pendingIntent)
+        startForeground(1, notification.build())
```
Comment (EntyR) [inline app/src/main/java/com/example/wearableapp/presentation/service/HeartRateSensorService.kt:62]: Please use [ongoing notification](https://developer.android.com/codelabs/ongoing-activity#4) here

Comment (EntyR) [inline app/src/main/java/com/example/wearableapp/presentation/service/HeartRateSensorService.kt:62]: Please use [ongoing notification](https://developer.android.com/codelabs/ongoing-activity#4) here

File: app/src/main/java/com/example/wearableapp/presentation/ui/ExportHeartRateActivity.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        binding = ActivityExportHeartRateBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setListeners()
+    }
+
+    private fun setListeners() {
+        binding.tvHeartExConfirmId.setOnClickListener(this)
+    }
+
+    override fun onClick(v: View?) {
```
Comment (EntyR) [inline app/src/main/java/com/example/wearableapp/presentation/ui/ExportHeartRateActivity.kt]: You can set listeners for every element with binding like this `binding.tvHeartExConfirmId.setOnClickListener { code }`

it's much cleaner this way

Comment (EntyR) [inline app/src/main/java/com/example/wearableapp/presentation/ui/ExportHeartRateActivity.kt]: You can set listeners for every element with binding like this `binding.tvHeartExConfirmId.setOnClickListener { code }`

it's much cleaner this way

File: app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    private fun stopMeasureHRView(){
+        stopAnimation()
+        binding.clHeartStartId.visibility = View.VISIBLE
+        binding.clHeartStopId.visibility = View.GONE
+        mSensorManager?.unregisterListener(this)
+    }
+
+    private fun getSensorData() {
```
Comment (EntyR) [inline app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt]: please move all sensor operation into service, we should be able to listen for events in ambient mode too

**Review by EntyR — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt (hunk 1 above)
Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt]: Yes i can . If we change emulator heart sensor movement we are getting heart rate event .

Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt]: Yes i can get it. If we change emulator heart sensor movement we are getting heart rate event .

File: app/src/main/java/com/example/wearableapp/presentation/service/HeartRateSensorService.kt (hunk 2 above)
Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/service/HeartRateSensorService.kt:62]: set ongoing flag as true

Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/service/HeartRateSensorService.kt:62]: set ongoing flag as true

File: app/src/main/java/com/example/wearableapp/presentation/ui/ExportHeartRateActivity.kt (hunk 3 above)
Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/ui/ExportHeartRateActivity.kt]: Resolved

Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/ui/ExportHeartRateActivity.kt]: Resolved

File: app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt (hunk 4 above)
Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt]: working on it

File: app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt (hunk 1 above)
Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt]: Done

File: app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt (hunk 4 above)
Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt]: moved sensor operator to service class

**Review by AlexeyKolosov — APPROVED** [review]

**Pull request merged**